packages/mui-material/src/styles/styled.js

```diff
@@ -1,11 +1,9 @@
 'use client';
-import createStyled, { shouldForwardProp } from '@mui/system/createStyled';
+import createStyled from '@mui/system/createStyled';
 import defaultTheme from './defaultTheme';
 import THEME_ID from './identifier';
-
-export const rootShouldForwardProp = (prop) => shouldForwardProp(prop) && prop !== 'classes';
-
-export const slotShouldForwardProp = shouldForwardProp;
+export { default as slotShouldForwardProp } from './slotShouldForwardProp';
```

Comment:
I had to extract these to separate files, otherwise we were ending up with importing from the styled.js, which resulted with the error:
```
/Users/marijanajdova/workspace/mui/apps/node_modules/.pnpm/@wyw-in-js+transform@0.5.0_typescript@5.3.3/node_modules/@wyw-in-js/transform/lib/module.js:223
      throw new EvalError(`${e.message} in${this.callstack.join('\n| ')}\n`);
      ^

EvalError: _systemDefaultTheme is not defined in/Users/marijanajdova/workspace/mui/packages/mui-system/build/createStyled.js
| /Users/marijanajdova/workspace/mui/packages/mui-material/build/styles/slotShouldForwardProp.js
| /Users/marijanajdova/workspace/mui/packages/mui-material/build/Slider/Slider.js
```

Once the PR is merged, I will update the umbrella issue to reflect this change, naming:

```diff
-import { slotShouldForwardProp}  from '../styles/styled';
+import slotShouldForwardProp from '../styles/slotShouldForwardProp';
```

and

```diff
-import { rootShouldForwardProp}  from '../styles/styled';
+import rootShouldForwardProp from '../styles/rootShouldForwardProp';
```

cc @brijeshb42, @siriwatknp

Reply:
I found a different fix in https://github.com/mui/material-ui/pull/41378/files#diff-23f194f4651836b6ce3d86b341074002f86e1bfcf3958c665fdf25a18aecfafb but I think we should have both.